src/Common/ProxyConfigurationResolver.h

```diff
@@ -9,9 +9,41 @@ struct ProxyConfigurationResolver
 {
     using Protocol = ProxyConfiguration::Protocol;
 
+    enum class ConnectProtocolPolicy
+    {
+        DEFAULT,
+        FORCE_OFF,
+        FORCE_ON
+    };
+
+    explicit ProxyConfigurationResolver(Protocol request_protocol_, ConnectProtocolPolicy connect_protocol_policy_ = ConnectProtocolPolicy::DEFAULT)
+        : request_protocol(request_protocol_), connect_protocol_policy(connect_protocol_policy_)
+    {
+    }
+
     virtual ~ProxyConfigurationResolver() = default;
     virtual ProxyConfiguration resolve() = 0;
     virtual void errorReport(const ProxyConfiguration & config) = 0;
+
+protected:
+    Protocol request_protocol;
+
+    // TODO think about any case
```

Comment:
`Protocol::ANY` is only used for old settings format. When parsing old settings format, `use_connect_protocol` is not read. Therefore, it is always `ConnectProtocolPolicy::DEFAULT`.

Reply:
Hm.. that might be a problem?